lib/src/deriv_chart/interactive_layer/widgets/selected_drawing_floating_menu.dart

```diff
@@ -0,0 +1,174 @@
+    _controller = widget.interactiveLayerBehaviour.controller;
+    _scaleAnimation = CurvedAnimation(
+      parent: Tween<double>(begin: 0.8, end: 1)
+          .animate(widget.interactiveLayerBehaviour.stateChangeController),
+      curve: Curves.easeOut,
+    );
+
+    _fadeAnimation = CurvedAnimation(
+      parent: widget.interactiveLayerBehaviour.stateChangeController,
+      curve: Curves.easeOut,
+    );
+
+    // Schedule a post-frame callback to get the menu size
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      _updateMenuSize();
+    });
+  }
+
+  void _updateMenuSize() {
+    // Get the size of this menu widget
+    final RenderBox? renderBox = context.findRenderObject() as RenderBox?;
+    if (renderBox != null) {
+      setState(() {
+        _menuSize = renderBox.size;
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // Use MediaQuery to get the screen size as a fallback
+    final screenSize = MediaQuery.of(context).size;
+
+    return Positioned(
+      left: _controller.floatingMenuPosition.dx,
+      top: _controller.floatingMenuPosition.dy,
+      child: GestureDetector(
+        behavior: HitTestBehavior.opaque,
+        onPanUpdate: (details) {
+          // Calculate new position
+          final newPosition = _controller.floatingMenuPosition + details.delta;
+
+          // Update menu size if not already set
+          if (_menuSize == Size.zero) {
+            _updateMenuSize();
+          }
+
+          // Find the nearest ancestor that provides size constraints
+          final RenderBox? ancestorBox =
+              context.findRenderObject()?.parent as RenderBox?;
+          final Size parentSize = ancestorBox?.size ?? screenSize;
+
+          // Constrain the position to keep the menu within the parent boundaries
+          final constrainedX =
+              newPosition.dx.clamp(0.0, parentSize.width - _menuSize.width);
+          final constrainedY =
+              newPosition.dy.clamp(0.0, parentSize.height - _menuSize.height);
+
+          _controller.floatingMenuPosition = Offset(constrainedX, constrainedY);
+          setState(() {});
+        },
+        child: AnimatedBuilder(
+          animation: widget.interactiveLayerBehaviour.stateChangeController,
+          builder: (_, child) => FadeTransition(
+            opacity: _fadeAnimation,
+            child: ScaleTransition(
+              scale: _scaleAnimation,
+              alignment: Alignment.topCenter,
+              child: child,
+            ),
+          ),
+          child: Container(
+            decoration: BoxDecoration(
+              // TODO(NA): use a color from theme when the theme specification in
+              // design documents has included the color for this menu.
+              color: CoreDesignTokens.coreColorSolidSlate1100,
+              borderRadius: BorderRadius.circular(8),
+            ),
+            padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
```

Comment:
To adjust with design
```suggestion
            padding: const EdgeInsets.only(right: 8, top: 4, bottom: 4),
```